packages/coinstac-simulator/src/boot-client.js

```diff
@@ -13,44 +13,72 @@ const User = common.models.User;
 const stubComputationToRegistry = require('./stub-computation-to-registry');
 const LocalPipelineRunnerPool = common.models.pipeline.runner.pool.LocalPipelineRunnerPool;
 const RemoteComputationResult = common.models.computation.RemoteComputationResult;
-const logger = require('./utils/logger');
+const { getChildProcessLogger } = require('./utils/logging');
 const retry = require('retry');
 
+const logger = getChildProcessLogger();
+
+/**
+ * Placeholder for user's data. This is used with
+ * @type {Object}
+ */
+let userData = {
+  kickoff: true,
+};
+
+let initiate = false;
 let pool;
-let decl;
 let username;
 
-const boot = function boot(opts) {
-  if (!opts.declPath || !opts.username) {
-    throw new ReferenceError('missing decl or username');
+
+/**
+ * Boot client.
+ *
+ * @param {Object} params
+ * @param {string} params.computationPath
+ * @param {Object} [params.data]
+ * @param {boolean} [params.initiate=false] Whether the process should initiate
+ * the computation run.
+ * @param {string} params.username
+ * @returns {Promise}
+ */
+function boot({
+  computationPath,
+  data,
+  initiate: init,
+  username: uname,
```

Comment:
Something we haven’t discussed: which versions of Node.js we should target. I think we should target LTS and test against it, so we might have to drop some niceties like destructuring.